tests/test_lane.py

```diff
@@ -296,986 +293,1083 @@ def facetify(collection=Facets.COLLECTION_FULL,
+        eq_(False, pagination.has_next_page)
 
-        # Because this lane has no list-related information, the
-        # RuntimeError shouldn't pop up at all.
-        lane = SetCustomListErrorLane(self._db, self._default_library, self._str)
+        # When the database is updated, pagination knows.
+        for num in range(3):
+            self._work()
+        pagination.apply(query)
+        eq_(True, pagination.has_next_page)
 
-        # The minute we put in some list information, it does!
-        assert_raises(
-            RuntimeError, SetCustomListErrorLane, self._db, self._default_library,
-            self._str, list_data_source=DataSource.NYT
-        )
+        # Even when the query ends at the same size as a page, all is well.
+        pagination.offset = 4
+        eq_(False, pagination.has_next_page)
 
-        # It can be a DataSource, or a CustomList identifier. World == oyster.
-        assert_raises(
-            RuntimeError, SetCustomListErrorLane, self._db, self._default_library,
-            self._str, list_identifier=u"Staff Picks"
-        )
 
+class MockFeaturedWorks(object):
+    """A mock WorkList that mocks featured_works()."""
 
-class TestLanes(DatabaseTest):
+    def __init__(self):
+        self._featured_works = []
+        self.visible = True
+
+    def queue_featured_works(self, works):
+        """Set the next return value for featured_works()."""
+        self._featured_works.append(works)
 
-    def test_all_matching_genres(self):
-        fantasy, ig = Genre.lookup(self._db, classifier.Fantasy)
-        cooking, ig = Genre.lookup(self._db, classifier.Cooking)
-        matches = Lane.all_matching_genres(self._db, [fantasy, cooking])
-        names = sorted([x.name for x in matches])
-        eq_(
-            [u'Cooking', u'Epic Fantasy', u'Fantasy', u'Historical Fantasy', 
-             u'Urban Fantasy'], 
-            names
-        )
+    def featured_works(self, *args, **kwargs):
+        try:
+            return self._featured_works.pop(0)
+        except IndexError:
+            return []
 
-    def test_nonexistent_list_raises_exception(self):
-        assert_raises(
-            UndefinedLane, Lane, self._db, self._default_library, 
-            u"This Will Fail", list_identifier=u"No Such List"
-        )
+class MockWork(object):
+    """Acts as a Work or a MaterializedWork interchangeably."""
+    def __init__(self, id):
+        self.id = id
+        self.works_id = id
+
+class MockWorks(WorkList):
+    """A WorkList that mocks works() but not featured_works()."""
+
+    def __init__(self):
+        self.reset()
+
+    def reset(self):
+        self._works = []
+        self.works_calls = []
+
+    def queue_works(self, works):
+        """Set the next return value for works()."""
+        self._works.append(works)
+
+    def queue_featured_works(self, works):
+        """Set the next return value for featured_works()."""
+        self._featured_works.append(works)
```

Comment:
I thought this class didn't mock featured_works?

Reply:
Well, you've certainly got me there...